tasks/skynet/src/main/java/com/innowise/service/Factory.java

```diff
@@ -0,0 +1,108 @@
+package com.innowise.service;
+
+import com.innowise.model.RobotPart;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.BlockingDeque;
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.stream.Collectors;
+
+/**
+ * The Factory class, which produces parts. It runs in its own thread.
+ * This class acts as the "Producer" in the simulation.
+ */
+public class Factory implements Runnable {
+
+    private static final Logger LOGGER = LogManager.getLogger(Factory.class.getName());
+
+    /**
+     * The storage for parts. Using LinkedBlockingDeque as it's a thread-safe
+     * collection suitable for producer-consumer scenarios.
+     */
+    private final BlockingDeque<RobotPart> storage = new LinkedBlockingDeque<>();
+    private final Random random = new Random();
+    private final CyclicBarrier barrier;
+    private final int simulationDays;
+
+    /**
+     * Constructs a new Factory.
+     *
+     * @param barrier        The CyclicBarrier used to synchronize simulation steps (day/night).
+     * @param simulationDays The total number of days the simulation will run.
+     */
+    public Factory(CyclicBarrier barrier, int simulationDays) {
+        this.barrier = barrier;
+        this.simulationDays = simulationDays;
+    }
+
+    @Override
+    public void run() {
+        for (int day = 1; day <= simulationDays; day++) {
```

Comment:
The requirements state that factions go to the factory "at night", but there's no clear day/night distinction in the code. The CyclicBarrier is used for synchronization, but the code doesn't explicitly model day and night phases